avahi-core/server.c

```diff
@@ -1658,8 +1682,35 @@ AvahiServerConfig* avahi_server_config_copy(AvahiServerConfig *ret, const AvahiS
     ret->browse_domains = browse;
     ret->allow_interfaces = allow;
     ret->deny_interfaces = deny;
+    ret->allow_addresses = allow_addresses;
+    ret->deny_addresses = deny_addresses;
 
     return ret;
+
+clean:
+    if (deny_addresses) {
```

Comment:
Here we also want to reuse common code ``avahi_address_list_free(deny_addresses)``.